File: src/main/java/me/Oxygen/modules/render/ClickGui.java
package me.Oxygen.modules.render;

import org.lwjgl.input.Keyboard;

import me.Oxygen.Oxygen;
import me.Oxygen.modules.Category;
import me.Oxygen.modules.Module;
import me.Oxygen.modules.ModuleRegister;
import me.Oxygen.ui.clickgui4.WorstClickGui;
import me.Oxygen.value.Value;

@ModuleRegister(name = "ClickGui", category = Category.RENDER, keybind = Keyboard.KEY_RSHIFT)
public class ClickGui extends Module{
	
	private Value<String> mode = new Value<String>("ClickGui", "Mode", 0);

	public ClickGui() {
		mode.addValue("Normal");
		mode.addValue("Jello");
		mode.addValue("Remix");
		mode.addValue("Power");
	}
	
	@Override
	public void onEnable() {
		switch(mode.getModeAt(mode.getCurrentMode())) {
		case "Normal" :
			this.mc.displayGuiScreen(Oxygen.INSTANCE.clickgui);
			this.set(false);
			break;
		case "Jello" :
			this.mc.displayGuiScreen(Oxygen.INSTANCE.crink);
			this.set(false);
			break;
		case "Remix" :
			this.mc.displayGuiScreen(new me.Oxygen.ui.clickgui2.ClickGui());
			this.set(false);
			break;
		case "Power" :
			this.mc.displayGuiScreen(new WorstClickGui());
			this.set(false);
			break;
		}
		super.onEnable();
	}

}

File: src/main/java/cn/rainbow/oxygen/module/modules/render/ClickGui.kt
package cn.rainbow.oxygen.module.modules.render

import cn.rainbow.oxygen.Oxygen
import cn.rainbow.oxygen.module.Category
import cn.rainbow.oxygen.module.Module
import cn.rainbow.oxygen.module.ModuleInfo
import cn.rainbow.oxygen.module.setting.ModeValue
import org.lwjgl.input.Keyboard

@ModuleInfo(name = "ClickGui", category = Category.Render)
class ClickGui: Module() {

    companion object {
        @JvmStatic
        val mode = ModeValue("Color", "Dark")
    }

    init {
        mode.addValue("Light")
        mode.addValue("Dark")
        this.keyCode = Keyboard.KEY_RSHIFT
    }

    override fun onEnable() {
        mc.displayGuiScreen(Oxygen.INSTANCE.briskGUI)
        this.enabled = false
    }

    override fun onDisable() {}
}